cmake/compilers.cmake

```diff
@@ -8,32 +8,19 @@ if (DEFINED MSVC_VERSION)
     message(STATUS "MSVC_VERSION: ${MSVC_VERSION}")
 endif()
 
-# Set variable for VS6 to handle special cases.
-if (DEFINED MSVC_VERSION AND MSVC_VERSION LESS 1300)
-    set(IS_VS6_BUILD TRUE)
-else()
-    set(IS_VS6_BUILD FALSE)
-endif()
-
 # Make release builds have debug information too.
 if(MSVC)
     # Create PDB for Release as long as debug info was generated during compile.
     string(APPEND CMAKE_EXE_LINKER_FLAGS_RELEASE " /DEBUG /OPT:REF /OPT:ICF")
     string(APPEND CMAKE_SHARED_LINKER_FLAGS_RELEASE " /DEBUG /OPT:REF /OPT:ICF")
+    # Multithreaded build.
+    add_compile_options(/MP)
+    # Enforce strict __cplusplus version
+    add_compile_options(/Zc:__cplusplus)
 else()
     # We go a bit wild here and assume any other compiler we are going to use supports -g for debug info.
-    string(APPEND CMAKE_CXX_FLAGS_RELEASE " -g")
-    string(APPEND CMAKE_C_FLAGS_RELEASE " -g")
+    add_compile_options(-g)
```

Comment:
I'd not do this. There is `CMAKE_BUILD_TYPE` for that.

Reply:
Build types all apply different optimisation, I assume you are referring to RelWithDebInfo which does not do what you expect. On GCC/Clang it applies -O2 optimisation rather than -O3 applied in Release configuration and is better thought of "debuggable with optimisation". On MSVC debug symbol generation has been decoupled from the build type leaving the build types largely only differentiated by the level of optimisation. Debug symbols are desirable these days in all configurations as they can be uploaded to crash analysis services and crash reports aggregated using libraries like Google's "crashpad".

Reply:
Not entirely convinced as its not optional for other build types, but we can revisit it later.

Reply:
My 2c is to not add any build options/optimizations in the cmake code, and add these option to cmake presets or a release generator build script.
You're right about `-O3` potentially introducing undefined behavior, but we should still keep it possible to build with `-O3`, if only to find issues.